src/GovUk.Education.ExploreEducationStatistics.Data.Model/Location.cs

```diff
@@ -11,8 +11,7 @@ public class Location
     {
         public Guid Id { get; set; }
 
-        // TODO EES-2776 SOW8 Make this non nullable when values have been migrated
-        public GeographicLevel? GeographicLevel { get; set; }
+        public GeographicLevel GeographicLevel { get; set; }
```

Comment:
I guess in order for this to happen, we need to have decided how to deal with that case on Prod whereby 2 Observation rows reference the same Location row but for different GeographicLevels?

Reply:
I was planning to just sort it manually . Here's what the Observation rows and Location row currently look like.

(Fictitious values, abbreviated columns names, etc.)

**Observation:**

```
Id      SubjectId  LocationId   GeographicLevel
===============================================
obs1    sub1       loc1         LA
obs2    sub1       loc1         LAD
```

**Location:**

New column `Location.GeorgraphicLevel` will have been added by this point from #3073.

```
Id    GeographicLevel   Country_Code    Country_Name    Region_Code    Region_Name    La_Code    La_Name    Lad_Code   Lad_Name
=================================================================================================================================
loc1  <null>            E92000001       England         E12000004      East Midlands  E06000015  Derby      E05000003  Derbyshire
```

We can just fix it with something like this:

```
INSERT INTO LOCATION (Id,GeographicLevel,Country_Code,Country_Name,LocalAuthority_Code,LocalAuthority_Name,LocalAuthorityDistrict_Code,LocalAuthorityDistrict_Name)
VALUES('loc2','LAD','E92000001','England','E12000004','East Midlands','E06000015','Derby','E05000003','Derbyshire');
`UPDATE Observation SET LocationId = 'loc2' WHERE Id = 'obs2';`
`UPDATE Location SET GeographicLevel = 'LA' Where Id = 'loc1';`
```

Re-running the stored procedure to copy the values across from Observation to Location to 'catch-up' would also have the same effect as the last `UPDATE` statement.

Ideally we will do all this and get rid of the uniqueness problem before this PR is merged so that when it is the querying and location lookup in the Importer works reliably for every possible case.

Reply:
This has been dealt with by EES-3010 now.